### PR TITLE
Tpetra,Teuchos: Fix #2567; Fix #1183

### LINKED ISSUE #1183: Teuchos: Remove Teuchos_ENABLE_LONG_LONG_INT option
@trilinos/teuchos 

The ability to disable `long long` support in Teuchos causes spurious issues like #1161.  The only reason not to allow use of `long long` is because of some Teuchos users who do not enable C++11 in their builds.  We need to respect these users, but it doesn't make sense to make `Teuchos_ENABLE_LONG_LONG_INT` a user-settable option.  If you have C++11, you should get `long long`, whether you want it or not.

### LINKED ISSUE #2567: Tpetra: Ignore Teuchos_ENABLE_LONG_LONG_INT
See PR #2547, Issue #2498, and PR #2499 for context.  Several years ago, the way to tell Tpetra to disable `GlobalOrdinal=long long` support, was to set `Teuchos_ENABLE_LONG_LONG_INT:BOOL=OFF`.  There was apparently no other way to do it.  That was bad.

Later on, we added proper Tpetra options for enabling or disabling `GlobalOrdinal=long long` support, in Tpetra.  However, users tend to keep and reuse CMake configuration options for Trilinos.  As a result, users kept expecting that `Teuchos_ENABLE_LONG_LONG_INT:BOOL=OFF` would work.  #2495 is an example of an issue that came up for this reason.

I worked around this with PR #2499, but the right approach is for Tpetra simply to ignore this CMake option.  Users should not set this Teuchos option any more!  Teuchos requires C++11 -- it has for a while -- and C++11 requires `long long`.  We shouldn't be afraid to remove old options. 

I would also like Teuchos to require that this option always be ON, but that's a separate issue.

@trilinos/tpetra

### PR DESCRIPTION
**Pull request by mhoemmen**
@trilinos/tpetra @trilinos/teuchos

## Description

Remove the `Teuchos_ENABLE_LONG_LONG_INT` option from Teuchos. 
 `HAVE_TEUCHOS_LONG_LONG_INT` is now always defined.

## Motivation and Context

Teuchos has assumed C++11 for years, so it's safe to require that `long long` exists.

## Related Issues

* Closes #1183, #2567

### CODE REVIEW

File: packages/teuchos/core/cmake/Teuchos_config.h.in [hunk 1]
```diff
@@ -109,7 +109,8 @@
 
 #cmakedefine HAVE_TEUCHOS_COMPLEX
 
-#cmakedefine HAVE_TEUCHOS_LONG_LONG_INT
+// This exists only for backwards compatibility.
+#define HAVE_TEUCHOS_LONG_LONG_INT 1
```
Comment (bartlettroscoe) [inline packages/teuchos/core/cmake/Teuchos_config.h.in]: Good

File: packages/tpetra/CMakeLists.txt [hunk 2]
```diff
@@ -610,16 +610,7 @@ GLOBAL_SET(HAVE_TPETRA_INST_INT_INT_DEFAULT ON)
...
-# This may change in the future, though.
-IF (DEFINED Teuchos_ENABLE_LONG_LONG_INT)
-  IF (NOT Teuchos_ENABLE_LONG_LONG_INT)
-    MESSAGE(WARNING "Do NOT set Teuchos_ENABLE_LONG_LONG_INT=OFF !  The default is ON.  We will VERY SOON require that this option be ON.  Don't set it explicitly!  Just use the default value.  See #1184: https://github.com/trilinos/Trilinos/issues/1184 and see also #1183: https://github.com/trilinos/Trilinos/issues/1183")
-  ENDIF ()
-  GLOBAL_SET(HAVE_TPETRA_INST_INT_LONG_LONG_DEFAULT ${Teuchos_ENABLE_LONG_LONG_INT})
-ELSE ()
-  GLOBAL_SET(HAVE_TPETRA_INST_INT_LONG_LONG_DEFAULT ON)
-ENDIF ()
+GLOBAL_SET(HAVE_TPETRA_INST_INT_LONG_LONG_DEFAULT ON)
```
Comment (bartlettroscoe) [inline packages/tpetra/CMakeLists.txt]: Good

File: packages/tpetra/doc/FAQ.txt [hunk 3]
```diff
@@ -392,28 +392,15 @@ GO types it enables by default.
 ** How does Tpetra pick which GO types to enable by default?
 
 By default, Tpetra currently enables ~GO = int~ and ~GO = long long~.
-Some build options affect Tpetra's default behavior; see below.
+If users enable ~GO = long~ (by setting ~Tpetra_INST_INT_LONG:BOOL=ON~
```
Comment (bartlettroscoe) [inline packages/tpetra/doc/FAQ.txt]: Yes

**Comment by mhoemmen** [comment]
@bartlettroscoe wrote:
> Not my place to approve this since I was not listed as one of the reviewers so this is just a comment.

I totally forgot to add you so please feel free to approve if you like it :-)  Thanks!

**Pull request merged**